// google-ads-node/build/src/index.js
"use strict";
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by gapic-generator-typescript. **
// ** https://github.com/googleapis/gapic-generator-typescript **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionGoalCampaignConfigServiceClient = exports.ConversionCustomVariableServiceClient = exports.ConversionAdjustmentUploadServiceClient = exports.ConversionActionServiceClient = exports.CampaignSharedSetServiceClient = exports.CampaignServiceClient = exports.CampaignLabelServiceClient = exports.CampaignGroupServiceClient = exports.CampaignFeedServiceClient = exports.CampaignExtensionSettingServiceClient = exports.CampaignDraftServiceClient = exports.CampaignCustomizerServiceClient = exports.CampaignCriterionServiceClient = exports.CampaignConversionGoalServiceClient = exports.CampaignBudgetServiceClient = exports.CampaignBidModifierServiceClient = exports.CampaignAssetSetServiceClient = exports.CampaignAssetServiceClient = exports.BillingSetupServiceClient = exports.BiddingStrategyServiceClient = exports.BiddingSeasonalityAdjustmentServiceClient = exports.BiddingDataExclusionServiceClient = exports.BatchJobServiceClient = exports.AudienceServiceClient = exports.AudienceInsightsServiceClient = exports.AssetSetServiceClient = exports.AssetSetAssetServiceClient = exports.AssetServiceClient = exports.AssetGroupSignalServiceClient = exports.AssetGroupServiceClient = exports.AssetGroupListingGroupFilterServiceClient = exports.AssetGroupAssetServiceClient = exports.AdServiceClient = exports.AdParameterServiceClient = exports.AdGroupServiceClient = exports.AdGroupLabelServiceClient = exports.AdGroupFeedServiceClient = exports.AdGroupExtensionSettingServiceClient = exports.AdGroupCustomizerServiceClient = exports.AdGroupCriterionServiceClient = exports.AdGroupCriterionLabelServiceClient = exports.AdGroupCriterionCustomizerServiceClient = exports.AdGroupBidModifierServiceClient = exports.AdGroupAssetSetServiceClient = exports.AdGroupAssetServiceClient = exports.AdGroupAdServiceClient = exports.AdGroupAdLabelServiceClient = exports.AccountLinkServiceClient = exports.AccountBudgetProposalServiceClient = exports.v12 = void 0;
exports.SharedSetServiceClient = exports.SharedCriterionServiceClient = exports.RemarketingActionServiceClient = exports.RecommendationServiceClient = exports.ReachPlanServiceClient = exports.PaymentsAccountServiceClient = exports.OfflineUserDataJobServiceClient = exports.MerchantCenterLinkServiceClient = exports.MediaFileServiceClient = exports.LabelServiceClient = exports.KeywordThemeConstantServiceClient = exports.KeywordPlanServiceClient = exports.KeywordPlanIdeaServiceClient = exports.KeywordPlanCampaignServiceClient = exports.KeywordPlanCampaignKeywordServiceClient = exports.KeywordPlanAdGroupServiceClient = exports.KeywordPlanAdGroupKeywordServiceClient = exports.InvoiceServiceClient = exports.GoogleAdsServiceClient = exports.GoogleAdsFieldServiceClient = exports.GeoTargetConstantServiceClient = exports.FeedServiceClient = exports.FeedMappingServiceClient = exports.FeedItemTargetServiceClient = exports.FeedItemSetServiceClient = exports.FeedItemSetLinkServiceClient = exports.FeedItemServiceClient = exports.ExtensionFeedItemServiceClient = exports.ExperimentServiceClient = exports.ExperimentArmServiceClient = exports.CustomizerAttributeServiceClient = exports.CustomInterestServiceClient = exports.CustomerUserAccessServiceClient = exports.CustomerUserAccessInvitationServiceClient = exports.CustomerServiceClient = exports.CustomerNegativeCriterionServiceClient = exports.CustomerManagerLinkServiceClient = exports.CustomerLabelServiceClient = exports.CustomerFeedServiceClient = exports.CustomerExtensionSettingServiceClient = exports.CustomerCustomizerServiceClient = exports.CustomerConversionGoalServiceClient = exports.CustomerClientLinkServiceClient = exports.CustomerAssetSetServiceClient = exports.CustomerAssetServiceClient = exports.CustomConversionGoalServiceClient = exports.CustomAudienceServiceClient = exports.ConversionValueRuleSetServiceClient = exports.ConversionValueRuleServiceClient = exports.ConversionUploadServiceClient = void 0;
exports.protos = exports.UserListServiceClient = exports.UserDataServiceClient = exports.ThirdPartyAppAnalyticsLinkServiceClient = exports.SmartCampaignSuggestServiceClient = exports.SmartCampaignSettingServiceClient = void 0;
const v12 = require("./v12");
exports.v12 = v12;
const AccountBudgetProposalServiceClient = v12.AccountBudgetProposalServiceClient;
exports.AccountBudgetProposalServiceClient = AccountBudgetProposalServiceClient;
const AccountLinkServiceClient = v12.AccountLinkServiceClient;
exports.AccountLinkServiceClient = AccountLinkServiceClient;
const AdGroupAdLabelServiceClient = v12.AdGroupAdLabelServiceClient;
exports.AdGroupAdLabelServiceClient = AdGroupAdLabelServiceClient;
const AdGroupAdServiceClient = v12.AdGroupAdServiceClient;
exports.AdGroupAdServiceClient = AdGroupAdServiceClient;
const AdGroupAssetServiceClient = v12.AdGroupAssetServiceClient;
exports.AdGroupAssetServiceClient = AdGroupAssetServiceClient;
const AdGroupAssetSetServiceClient = v12.AdGroupAssetSetServiceClient;
exports.AdGroupAssetSetServiceClient = AdGroupAssetSetServiceClient;
const AdGroupBidModifierServiceClient = v12.AdGroupBidModifierServiceClient;
exports.AdGroupBidModifierServiceClient = AdGroupBidModifierServiceClient;
const AdGroupCriterionCustomizerServiceClient = v12.AdGroupCriterionCustomizerServiceClient;
exports.AdGroupCriterionCustomizerServiceClient = AdGroupCriterionCustomizerServiceClient;
const AdGroupCriterionLabelServiceClient = v12.AdGroupCriterionLabelServiceClient;
exports.AdGroupCriterionLabelServiceClient = AdGroupCriterionLabelServiceClient;
const AdGroupCriterionServiceClient = v12.AdGroupCriterionServiceClient;
exports.AdGroupCriterionServiceClient = AdGroupCriterionServiceClient;
const AdGroupCustomizerServiceClient = v12.AdGroupCustomizerServiceClient;
exports.AdGroupCustomizerServiceClient = AdGroupCustomizerServiceClient;
const AdGroupExtensionSettingServiceClient = v12.AdGroupExtensionSettingServiceClient;
exports.AdGroupExtensionSettingServiceClient = AdGroupExtensionSettingServiceClient;
const AdGroupFeedServiceClient = v12.AdGroupFeedServiceClient;
exports.AdGroupFeedServiceClient = AdGroupFeedServiceClient;
const AdGroupLabelServiceClient = v12.AdGroupLabelServiceClient;
exports.AdGroupLabelServiceClient = AdGroupLabelServiceClient;
const AdGroupServiceClient = v12.AdGroupServiceClient;
exports.AdGroupServiceClient = AdGroupServiceClient;
const AdParameterServiceClient = v12.AdParameterServiceClient;
exports.AdParameterServiceClient = AdParameterServiceClient;
const AdServiceClient = v12.AdServiceClient;
exports.AdServiceClient = AdServiceClient;
const AssetGroupAssetServiceClient = v12.AssetGroupAssetServiceClient;
exports.AssetGroupAssetServiceClient = AssetGroupAssetServiceClient;
const AssetGroupListingGroupFilterServiceClient = v12.AssetGroupListingGroupFilterServiceClient;
exports.AssetGroupListingGroupFilterServiceClient = AssetGroupListingGroupFilterServiceClient;
const AssetGroupServiceClient = v12.AssetGroupServiceClient;
exports.AssetGroupServiceClient = AssetGroupServiceClient;
const AssetGroupSignalServiceClient = v12.AssetGroupSignalServiceClient;
exports.AssetGroupSignalServiceClient = AssetGroupSignalServiceClient;
const AssetServiceClient = v12.AssetServiceClient;
exports.AssetServiceClient = AssetServiceClient;
const AssetSetAssetServiceClient = v12.AssetSetAssetServiceClient;
exports.AssetSetAssetServiceClient = AssetSetAssetServiceClient;
const AssetSetServiceClient = v12.AssetSetServiceClient;
exports.AssetSetServiceClient = AssetSetServiceClient;
const AudienceInsightsServiceClient = v12.AudienceInsightsServiceClient;
exports.AudienceInsightsServiceClient = AudienceInsightsServiceClient;
const AudienceServiceClient = v12.AudienceServiceClient;
exports.AudienceServiceClient = AudienceServiceClient;
const BatchJobServiceClient = v12.BatchJobServiceClient;
exports.BatchJobServiceClient = BatchJobServiceClient;
const BiddingDataExclusionServiceClient = v12.BiddingDataExclusionServiceClient;
exports.BiddingDataExclusionServiceClient = BiddingDataExclusionServiceClient;
const BiddingSeasonalityAdjustmentServiceClient = v12.BiddingSeasonalityAdjustmentServiceClient;
exports.BiddingSeasonalityAdjustmentServiceClient = BiddingSeasonalityAdjustmentServiceClient;
const BiddingStrategyServiceClient = v12.BiddingStrategyServiceClient;
exports.BiddingStrategyServiceClient = BiddingStrategyServiceClient;
const BillingSetupServiceClient = v12.BillingSetupServiceClient;
exports.BillingSetupServiceClient = BillingSetupServiceClient;
const CampaignAssetServiceClient = v12.CampaignAssetServiceClient;
exports.CampaignAssetServiceClient = CampaignAssetServiceClient;
const CampaignAssetSetServiceClient = v12.CampaignAssetSetServiceClient;
exports.CampaignAssetSetServiceClient = CampaignAssetSetServiceClient;
const CampaignBidModifierServiceClient = v12.CampaignBidModifierServiceClient;
exports.CampaignBidModifierServiceClient = CampaignBidModifierServiceClient;
const CampaignBudgetServiceClient = v12.CampaignBudgetServiceClient;
exports.CampaignBudgetServiceClient = CampaignBudgetServiceClient;
const CampaignConversionGoalServiceClient = v12.CampaignConversionGoalServiceClient;
exports.CampaignConversionGoalServiceClient = CampaignConversionGoalServiceClient;
const CampaignCriterionServiceClient = v12.CampaignCriterionServiceClient;
exports.CampaignCriterionServiceClient = CampaignCriterionServiceClient;
const CampaignCustomizerServiceClient = v12.CampaignCustomizerServiceClient;
exports.CampaignCustomizerServiceClient = CampaignCustomizerServiceClient;
const CampaignDraftServiceClient = v12.CampaignDraftServiceClient;
exports.CampaignDraftServiceClient = CampaignDraftServiceClient;
const CampaignExtensionSettingServiceClient = v12.CampaignExtensionSettingServiceClient;
exports.CampaignExtensionSettingServiceClient = CampaignExtensionSettingServiceClient;
const CampaignFeedServiceClient = v12.CampaignFeedServiceClient;
exports.CampaignFeedServiceClient = CampaignFeedServiceClient;
const CampaignGroupServiceClient = v12.CampaignGroupServiceClient;
exports.CampaignGroupServiceClient = CampaignGroupServiceClient;
const CampaignLabelServiceClient = v12.CampaignLabelServiceClient;
exports.CampaignLabelServiceClient = CampaignLabelServiceClient;
const CampaignServiceClient = v12.CampaignServiceClient;
exports.CampaignServiceClient = CampaignServiceClient;
const CampaignSharedSetServiceClient = v12.CampaignSharedSetServiceClient;
exports.CampaignSharedSetServiceClient = CampaignSharedSetServiceClient;
const ConversionActionServiceClient = v12.ConversionActionServiceClient;
exports.ConversionActionServiceClient = ConversionActionServiceClient;
const ConversionAdjustmentUploadServiceClient = v12.ConversionAdjustmentUploadServiceClient;
exports.ConversionAdjustmentUploadServiceClient = ConversionAdjustmentUploadServiceClient;
const ConversionCustomVariableServiceClient = v12.ConversionCustomVariableServiceClient;
exports.ConversionCustomVariableServiceClient = ConversionCustomVariableServiceClient;
const ConversionGoalCampaignConfigServiceClient = v12.ConversionGoalCampaignConfigServiceClient;
exports.ConversionGoalCampaignConfigServiceClient = ConversionGoalCampaignConfigServiceClient;
const ConversionUploadServiceClient = v12.ConversionUploadServiceClient;
exports.ConversionUploadServiceClient = ConversionUploadServiceClient;
const ConversionValueRuleServiceClient = v12.ConversionValueRuleServiceClient;
exports.ConversionValueRuleServiceClient = ConversionValueRuleServiceClient;
const ConversionValueRuleSetServiceClient = v12.ConversionValueRuleSetServiceClient;
exports.ConversionValueRuleSetServiceClient = ConversionValueRuleSetServiceClient;
const CustomAudienceServiceClient = v12.CustomAudienceServiceClient;
exports.CustomAudienceServiceClient = CustomAudienceServiceClient;
const CustomConversionGoalServiceClient = v12.CustomConversionGoalServiceClient;
exports.CustomConversionGoalServiceClient = CustomConversionGoalServiceClient;
const CustomerAssetServiceClient = v12.CustomerAssetServiceClient;
exports.CustomerAssetServiceClient = CustomerAssetServiceClient;
const CustomerAssetSetServiceClient = v12.CustomerAssetSetServiceClient;
exports.CustomerAssetSetServiceClient = CustomerAssetSetServiceClient;
const CustomerClientLinkServiceClient = v12.CustomerClientLinkServiceClient;
exports.CustomerClientLinkServiceClient = CustomerClientLinkServiceClient;
const CustomerConversionGoalServiceClient = v12.CustomerConversionGoalServiceClient;
exports.CustomerConversionGoalServiceClient = CustomerConversionGoalServiceClient;
const CustomerCustomizerServiceClient = v12.CustomerCustomizerServiceClient;
exports.CustomerCustomizerServiceClient = CustomerCustomizerServiceClient;
const CustomerExtensionSettingServiceClient = v12.CustomerExtensionSettingServiceClient;
exports.CustomerExtensionSettingServiceClient = CustomerExtensionSettingServiceClient;
const CustomerFeedServiceClient = v12.CustomerFeedServiceClient;
exports.CustomerFeedServiceClient = CustomerFeedServiceClient;
const CustomerLabelServiceClient = v12.CustomerLabelServiceClient;
exports.CustomerLabelServiceClient = CustomerLabelServiceClient;
const CustomerManagerLinkServiceClient = v12.CustomerManagerLinkServiceClient;
exports.CustomerManagerLinkServiceClient = CustomerManagerLinkServiceClient;
const CustomerNegativeCriterionServiceClient = v12.CustomerNegativeCriterionServiceClient;
exports.CustomerNegativeCriterionServiceClient = CustomerNegativeCriterionServiceClient;
const CustomerServiceClient = v12.CustomerServiceClient;
exports.CustomerServiceClient = CustomerServiceClient;
const CustomerUserAccessInvitationServiceClient = v12.CustomerUserAccessInvitationServiceClient;
exports.CustomerUserAccessInvitationServiceClient = CustomerUserAccessInvitationServiceClient;
const CustomerUserAccessServiceClient = v12.CustomerUserAccessServiceClient;
exports.CustomerUserAccessServiceClient = CustomerUserAccessServiceClient;
const CustomInterestServiceClient = v12.CustomInterestServiceClient;
exports.CustomInterestServiceClient = CustomInterestServiceClient;
const CustomizerAttributeServiceClient = v12.CustomizerAttributeServiceClient;
exports.CustomizerAttributeServiceClient = CustomizerAttributeServiceClient;
const ExperimentArmServiceClient = v12.ExperimentArmServiceClient;
exports.ExperimentArmServiceClient = ExperimentArmServiceClient;
const ExperimentServiceClient = v12.ExperimentServiceClient;
exports.ExperimentServiceClient = ExperimentServiceClient;
const ExtensionFeedItemServiceClient = v12.ExtensionFeedItemServiceClient;
exports.ExtensionFeedItemServiceClient = ExtensionFeedItemServiceClient;
const FeedItemServiceClient = v12.FeedItemServiceClient;
exports.FeedItemServiceClient = FeedItemServiceClient;
const FeedItemSetLinkServiceClient = v12.FeedItemSetLinkServiceClient;
exports.FeedItemSetLinkServiceClient = FeedItemSetLinkServiceClient;
const FeedItemSetServiceClient = v12.FeedItemSetServiceClient;
exports.FeedItemSetServiceClient = FeedItemSetServiceClient;
const FeedItemTargetServiceClient = v12.FeedItemTargetServiceClient;
exports.FeedItemTargetServiceClient = FeedItemTargetServiceClient;
const FeedMappingServiceClient = v12.FeedMappingServiceClient;
exports.FeedMappingServiceClient = FeedMappingServiceClient;
const FeedServiceClient = v12.FeedServiceClient;
exports.FeedServiceClient = FeedServiceClient;
const GeoTargetConstantServiceClient = v12.GeoTargetConstantServiceClient;
exports.GeoTargetConstantServiceClient = GeoTargetConstantServiceClient;
const GoogleAdsFieldServiceClient = v12.GoogleAdsFieldServiceClient;
exports.GoogleAdsFieldServiceClient = GoogleAdsFieldServiceClient;
const GoogleAdsServiceClient = v12.GoogleAdsServiceClient;
exports.GoogleAdsServiceClient = GoogleAdsServiceClient;
const InvoiceServiceClient = v12.InvoiceServiceClient;
exports.InvoiceServiceClient = InvoiceServiceClient;
const KeywordPlanAdGroupKeywordServiceClient = v12.KeywordPlanAdGroupKeywordServiceClient;
exports.KeywordPlanAdGroupKeywordServiceClient = KeywordPlanAdGroupKeywordServiceClient;
const KeywordPlanAdGroupServiceClient = v12.KeywordPlanAdGroupServiceClient;
exports.KeywordPlanAdGroupServiceClient = KeywordPlanAdGroupServiceClient;
const KeywordPlanCampaignKeywordServiceClient = v12.KeywordPlanCampaignKeywordServiceClient;
exports.KeywordPlanCampaignKeywordServiceClient = KeywordPlanCampaignKeywordServiceClient;
const KeywordPlanCampaignServiceClient = v12.KeywordPlanCampaignServiceClient;
exports.KeywordPlanCampaignServiceClient = KeywordPlanCampaignServiceClient;
const KeywordPlanIdeaServiceClient = v12.KeywordPlanIdeaServiceClient;
exports.KeywordPlanIdeaServiceClient = KeywordPlanIdeaServiceClient;
const KeywordPlanServiceClient = v12.KeywordPlanServiceClient;
exports.KeywordPlanServiceClient = KeywordPlanServiceClient;
const KeywordThemeConstantServiceClient = v12.KeywordThemeConstantServiceClient;
exports.KeywordThemeConstantServiceClient = KeywordThemeConstantServiceClient;
const LabelServiceClient = v12.LabelServiceClient;
exports.LabelServiceClient = LabelServiceClient;
const MediaFileServiceClient = v12.MediaFileServiceClient;
exports.MediaFileServiceClient = MediaFileServiceClient;
const MerchantCenterLinkServiceClient = v12.MerchantCenterLinkServiceClient;
exports.MerchantCenterLinkServiceClient = MerchantCenterLinkServiceClient;
const OfflineUserDataJobServiceClient = v12.OfflineUserDataJobServiceClient;
exports.OfflineUserDataJobServiceClient = OfflineUserDataJobServiceClient;
const PaymentsAccountServiceClient = v12.PaymentsAccountServiceClient;
exports.PaymentsAccountServiceClient = PaymentsAccountServiceClient;
const ReachPlanServiceClient = v12.ReachPlanServiceClient;
exports.ReachPlanServiceClient = ReachPlanServiceClient;
const RecommendationServiceClient = v12.RecommendationServiceClient;
exports.RecommendationServiceClient = RecommendationServiceClient;
const RemarketingActionServiceClient = v12.RemarketingActionServiceClient;
exports.RemarketingActionServiceClient = RemarketingActionServiceClient;
const SharedCriterionServiceClient = v12.SharedCriterionServiceClient;
exports.SharedCriterionServiceClient = SharedCriterionServiceClient;
const SharedSetServiceClient = v12.SharedSetServiceClient;
exports.SharedSetServiceClient = SharedSetServiceClient;
const SmartCampaignSettingServiceClient = v12.SmartCampaignSettingServiceClient;
exports.SmartCampaignSettingServiceClient = SmartCampaignSettingServiceClient;
const SmartCampaignSuggestServiceClient = v12.SmartCampaignSuggestServiceClient;
exports.SmartCampaignSuggestServiceClient = SmartCampaignSuggestServiceClient;
const ThirdPartyAppAnalyticsLinkServiceClient = v12.ThirdPartyAppAnalyticsLinkServiceClient;
exports.ThirdPartyAppAnalyticsLinkServiceClient = ThirdPartyAppAnalyticsLinkServiceClient;
const UserDataServiceClient = v12.UserDataServiceClient;
exports.UserDataServiceClient = UserDataServiceClient;
const UserListServiceClient = v12.UserListServiceClient;
exports.UserListServiceClient = UserListServiceClient;
exports.default = { v12, AccountBudgetProposalServiceClient, AccountLinkServiceClient, AdGroupAdLabelServiceClient, AdGroupAdServiceClient, AdGroupAssetServiceClient, AdGroupAssetSetServiceClient, AdGroupBidModifierServiceClient, AdGroupCriterionCustomizerServiceClient, AdGroupCriterionLabelServiceClient, AdGroupCriterionServiceClient, AdGroupCustomizerServiceClient, AdGroupExtensionSettingServiceClient, AdGroupFeedServiceClient, AdGroupLabelServiceClient, AdGroupServiceClient, AdParameterServiceClient, AdServiceClient, AssetGroupAssetServiceClient, AssetGroupListingGroupFilterServiceClient, AssetGroupServiceClient, AssetGroupSignalServiceClient, AssetServiceClient, AssetSetAssetServiceClient, AssetSetServiceClient, AudienceInsightsServiceClient, AudienceServiceClient, BatchJobServiceClient, BiddingDataExclusionServiceClient, BiddingSeasonalityAdjustmentServiceClient, BiddingStrategyServiceClient, BillingSetupServiceClient, CampaignAssetServiceClient, CampaignAssetSetServiceClient, CampaignBidModifierServiceClient, CampaignBudgetServiceClient, CampaignConversionGoalServiceClient, CampaignCriterionServiceClient, CampaignCustomizerServiceClient, CampaignDraftServiceClient, CampaignExtensionSettingServiceClient, CampaignFeedServiceClient, CampaignGroupServiceClient, CampaignLabelServiceClient, CampaignServiceClient, CampaignSharedSetServiceClient, ConversionActionServiceClient, ConversionAdjustmentUploadServiceClient, ConversionCustomVariableServiceClient, ConversionGoalCampaignConfigServiceClient, ConversionUploadServiceClient, ConversionValueRuleServiceClient, ConversionValueRuleSetServiceClient, CustomAudienceServiceClient, CustomConversionGoalServiceClient, CustomerAssetServiceClient, CustomerAssetSetServiceClient, CustomerClientLinkServiceClient, CustomerConversionGoalServiceClient, CustomerCustomizerServiceClient, CustomerExtensionSettingServiceClient, CustomerFeedServiceClient, CustomerLabelServiceClient, CustomerManagerLinkServiceClient, CustomerNegativeCriterionServiceClient, CustomerServiceClient, CustomerUserAccessInvitationServiceClient, CustomerUserAccessServiceClient, CustomInterestServiceClient, CustomizerAttributeServiceClient, ExperimentArmServiceClient, ExperimentServiceClient, ExtensionFeedItemServiceClient, FeedItemServiceClient, FeedItemSetLinkServiceClient, FeedItemSetServiceClient, FeedItemTargetServiceClient, FeedMappingServiceClient, FeedServiceClient, GeoTargetConstantServiceClient, GoogleAdsFieldServiceClient, GoogleAdsServiceClient, InvoiceServiceClient, KeywordPlanAdGroupKeywordServiceClient, KeywordPlanAdGroupServiceClient, KeywordPlanCampaignKeywordServiceClient, KeywordPlanCampaignServiceClient, KeywordPlanIdeaServiceClient, KeywordPlanServiceClient, KeywordThemeConstantServiceClient, LabelServiceClient, MediaFileServiceClient, MerchantCenterLinkServiceClient, OfflineUserDataJobServiceClient, PaymentsAccountServiceClient, ReachPlanServiceClient, RecommendationServiceClient, RemarketingActionServiceClient, SharedCriterionServiceClient, SharedSetServiceClient, SmartCampaignSettingServiceClient, SmartCampaignSuggestServiceClient, ThirdPartyAppAnalyticsLinkServiceClient, UserDataServiceClient, UserListServiceClient };
const protos = require("../protos/protos");
exports.protos = protos;
